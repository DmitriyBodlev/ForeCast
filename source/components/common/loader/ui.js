import styled, { keyframes } from 'react-emotion';

export const LoaderWrapper = styled.div`
  align-items: center;
  background: rgba(53, 53, 53, 0.5);
  display: flex;
  flex-direction: column;
  height: 100%;
  justify-content: center;
  position: fixed;
  top: 0;
  width: 100%;
`;

export const DimmerWrapper = styled.div`
  height: 100%;
  left: 0;
  top: 0;
  padding: 20px;
  position: absolute;
  width: 100%;
  z-index: 11;

  background-color: white;
`;

export const TitleRectangle = styled.div`
  height: 20px;
  margin: 20px 10px;
  width: 50px;

  background-color: grey;
`;

export const Rectangle = styled.div`
  height: 20px;
  margin-bottom: 20px;
  width: 100%;

  background-color: grey;
`;

export const TableRectangle = styled.div`
  margin-bottom: 20px;
  padding: 20px;
  width: 550px;

  background-color: lightgrey;
`;

export const squareRotate = keyframes`
  0% { 
    transform: perspective(120px) rotateX(0deg) rotateY(0deg);
  } 50% { 
    transform: perspective(120px) rotateX(-180.1deg) rotateY(0deg);
  } 100% { 
    transform: perspective(120px) rotateX(-180deg) rotateY(-179.9deg);
  }
`;

export const LoaderContent = styled.div`
  height: 40px;
  padding: 15px;
  border-radius: 2px;
  position: fixed;
  width: 40px;

  background-color: red;

  animation: ${squareRotate} 1.2s infinite ease-in-out;
  
`;
