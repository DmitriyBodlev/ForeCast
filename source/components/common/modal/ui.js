import styled, { css, keyframes } from 'react-emotion';

export const animation = (props: Object) => {
  const from = props.params.position.split('-')[0];

  const moves = {
    top: keyframes`
      0% { top: -${props.params.height}px }
      100% { top: 0 }
    `,
    bottom: keyframes`
      0% { bottom: -${props.params.height}px }
      100% { bottom: 0 }
    `,
  };

  return moves[from];
};

export const ModalWrapper = styled.div`
  position: fixed;
  width: 100%;
  height: 100%;
  top: 0;
  background: rgba(53, 53, 53, 0.5);
  display: flex;
  justify-content: center;
  align-items: center;
  flex-direction: column;
`;

export const Title = styled.span`
`;

export const ModalContent = styled.div`
  padding: 15px;
  position: fixed;
  border-radius: 2px;
  width: ${({params}: Object) => params.width}px;
  height: ${({params}: Object) => params.height}px;
  background-color: grey;
  box-shadow: 0 0 8px 1px black;
  animation: ${(props: Object) => animation(props)} 0.2s linear;
`;

export const Header = styled.div`
  font-weight: bold;
  color: black;
  font-size: 17px;
  align-items: center;
  display: inline-flex;
  width: 100%;
  justify-content: space-between;
`;

export const Footer = styled.div`
  position: absolute;
  bottom: 0px;
  width: 100%;
  border-top: 1px solid;
  height: 55px;
  padding: 10px 15px;
`;

const topLeft = css`
  top: 0;
  left: 0;
`;
const topCenter = css`
  top: 0;
  right: 50%;
  transform: translateX(50%);
`;
const topRight = css`
  top: 0;
  right: 0;
`;

const bottomLeft = css`
  bottom: 0;
  left: 0;
`;

const bottomCenter = css`
  bottom: 0;
  right: 50%;
  transform: translateX(50%);
`;

const bottomRight = css`
  bottom: 0;
  right: 0;
`;

const classStyles = {
  'top-left': topLeft,
  'top-center': topCenter,
  'top-right': topRight,
  'bottom-left': bottomLeft,
  'bottom-center': bottomCenter,
  'bottom-right': bottomRight,
};

export default classStyles;
