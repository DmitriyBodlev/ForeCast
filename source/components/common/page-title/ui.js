import { css } from 'react-emotion';

const wrapper = css`
  font-size: 20px;
  font-weight: 500;
  height: 60px;
  line-height: 60px;
  padding-left: 15px;

  background-color: white;
  color: black;
`;

const styles = {
  wrapper,
};

export default styles;
