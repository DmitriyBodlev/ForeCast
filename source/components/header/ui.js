import styled, { css } from 'react-emotion';

export const Divider = styled.span`
  display: inline-block;
  background-color: black;
  height: 32px;
  width: 1px;
  margin: auto 7px;
`;

export const HeaderWrapper = styled.div`
  background-color: grey;
  height: 100px;
  padding-right: 15px;
  width: 100%;
`;

export const Logo = styled.img`
  max-height: 70px;
  max-width: 100%;
  vertical-align: middle;
`;

const helperStyle = css`
  display: inline-block;
  height: 100%;
  vertical-align: middle;
`;

const logoStyle = css`
  float: left;
  margin-right: 3%;
  height: 70px;
  max-width: 15%;
  text-align: center;
  white-space: nowrap;
`;

const brandStyle = css`
  height: inherit;
  padding: 15px;
  position: relative;
`;

const textStyle = css`
  display: inline-block;
  height: 70px;
  line-height: 70px;
  text-align: left;
  vertical-align: middle;
  width: 39%;
`;

const menuStyle = css`
  display: flex;
  justify-content: flex-end;
  float: right;
  line-height: 15px;
  height: 30px;
  margin: 20px 0;
  width: 42%;
`;

const classStyle = {
  brand: brandStyle,
  helper: helperStyle,
  logo: logoStyle,
  menu: menuStyle,
  text: textStyle,
};

export default classStyle;
