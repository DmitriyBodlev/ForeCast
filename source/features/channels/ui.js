import styled from 'react-emotion';

export const BunnerWrapper = styled.div`
  background: #ffffff url(${(props: Object) => props.image}) no-repeat center center;
  background-size: cover;
  width: 100vw;
  height: 450px;
  display: flex;
  justify-content: center;
  align-items: center;
  & img {
    width: auto;
  }
`;

export const BlaBla = styled.div`
  height: 200px;
  width: 200px;
  background-color: black;
  color: white;
`;
