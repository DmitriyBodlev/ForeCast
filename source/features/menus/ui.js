import styled from 'react-emotion';
import { withTheme } from 'theming';

export const NavContainer = withTheme(styled.nav`
  box-shadow: 'inset 0 0 2px 1px rgba(217, 217, 217, 0.5)';
  display: flex;
  height: 40px;
  cursor: pointer;
  font-size: 12px;
  font-weight: 400;
  line-height: 40px;

  transition: margin-left .2s linear;

  margin-left: ${(props: Object) => props.openedSidebar ? '210px' : '0px'};
  background-color: ${({theme}: Object) => theme.menus.nav.bgColor};
  color: ${({theme}: Object) => theme.menus.nav.itemTextColor};
`);

const renderBackgroundIcon = (props: Object) => {
  if (props.currentTheme === 'dark') {
    return props.darkLink;
  }
  return props.lightLink;
};

export const NavItem = withTheme(styled.div`
  background-repeat: no-repeat;
  background-position-y: center;
  background-position-x: 15px;
  padding: 0 10px 0 45px;

  background-image: url(${(props: Object) => renderBackgroundIcon(props)});

  &:hover {
    background-color: ${({theme}: Object) => theme.menus.nav.bgColorHovered};
  }
`);

export const SidebarContainer = withTheme(styled.aside`
  box-shadow: 'inset 0 0 2px 1px rgba(217, 217, 217, 0.5)';
  cursor: default;
  font-size: 14px;
  height: calc(100vh - 87px);
  position: absolute;
  left: -210px;
  width: 210px;

  transition: left .2s linear;

  background-color: ${({theme}: Object) => theme.menus.sidebar.bgColor};
  color: ${({theme}: Object) => theme.menus.sidebar.itemTextColor};

  &.opened {
    left: 0px;
  }
`);

export const SidebarHeader = withTheme(styled.div`
  background-repeat: no-repeat;
  background-position-y: center;
  background-position-x: 180px;
  height: 40px;
  padding: 0 10px 0 45px;

  background-image: url('https://s3.amazonaws.com/amous-front-dev/icons/close.svg');

  &:hover {
    background-color: ${({theme}: Object) => theme.menus.sidebar.bgColorHovered};
  }
`);

const renderSidebarGroupWrapperBgColor = (props: Object) => {
  if (props.opened) {
    return props.theme.menus.sidebar.bgColorHovered;
  }
  return props.theme.menus.sidebar.bgColor;
};

export const SidebarGroupWrapper = withTheme(styled.div`
  background-color: ${(props: Object) => renderSidebarGroupWrapperBgColor(props)};
`);

export const SidebarGroupSubitemsWrapper = withTheme(styled.div`
`);

export const SidebarItem = withTheme(styled.div`
  background-repeat: no-repeat;
  background-position-y: center;
  background-position-x: 15px;
  cursor: pointer;
  height: 40px;
  line-height: 40px;
  padding: 0 10px 0 45px;

  background-image: url(${(props: Object) => renderBackgroundIcon(props)});
  font-weight: ${({opened}: Bool) => opened && 'bold'};
  color: ${(props: Object) => props.opened ? props.theme.menus.sidebar.itemTextColorHovered : props.theme.menus.sidebar.itemTextColor};

  &:hover {
    background-color: ${({theme}: Object) => theme.menus.sidebar.bgColorHovered};
    color: ${({theme}: Object) => theme.menus.sidebar.itemTextColorHovered};
  }
`);

export const SidebarItemRightPlusIcon = withTheme(styled.span`
  &::after { 
    content: "+";
    color: #ce2828;
    font-size: 20px;
    font-weight: 900;
    float: right;
  }
`);

const renderArrowIconRotate = ({opened}: boolean) => {
  if (opened) {
    return 'rotate(-45deg)';
  }
  return 'rotate(135deg)';
};

export const SidebarItemRightArrowIcon = withTheme(styled.span`
  &::after {
    border-top: 2px solid #ce2828;
    border-right: 2px solid #ce2828;
    content: '';
    display: block;
    float: right;
    margin-top: 15px;
    margin-right: 2px;
    width: 5px;
    height: 5px;

    transform: ${(props: Object) => renderArrowIconRotate(props)};
  }
`);
