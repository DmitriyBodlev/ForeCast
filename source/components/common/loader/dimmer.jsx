import React from 'react';
import { times, not } from 'ramda';
import { TableRectangle, TitleRectangle, Rectangle, DimmerWrapper} from './ui';

const renderDimmerRectangles = (rectangleCount: number) => (
  times((index: number) => <Rectangle key={index} />, rectangleCount)
);

const renderDimmerTable = ({tableCount, rectangleCount}: Object) => (
  times((index: number) => (
    <TableRectangle key={index}>
      {renderDimmerRectangles(rectangleCount)}
    </TableRectangle>
  ), tableCount)
);

export const DimmerComponent = (props: Object) => {

  return (
    <DimmerWrapper>
      <TitleRectangle />
      <Rectangle />
      { props.tableCount && renderDimmerTable(props)}
      {
        not(props.tableCount)
        && props.rectangleCount
        && renderDimmerRectangles(props.rectangleCount) }
    </DimmerWrapper>
  );
};

export default DimmerComponent;
