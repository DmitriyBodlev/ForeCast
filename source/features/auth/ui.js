import styled from 'react-emotion';
import { Form } from 'redux-form';
import { Button, LinkTo } from '../../ui';

const renderBackgroundLink = (props: Object) => {
  if (props.currentTheme === 'dark') {
    return 'https://s3.amazonaws.com/amous-front-dev/images/dark-background.svg';
  }
  return 'https://s3.amazonaws.com/amous-front-dev/images/light-background.svg';
};

export const FormWrapper = styled.div`
  align-items: center;
  background-image: url(${(props: Object) => renderBackgroundLink(props)});
  background-repeat: no-repeat;
  background-attachment: fixed;
  background-position: center; 
  background-size: cover;
  display: flex;
  overflow: auto;
  height: 100vh;
  width: 100vw;
`;

export const FormContainer = styled.div`
  margin-left: 25%;
  width: 50%;

  & input {
    height: 42px;

    &[type='text']:focus, &[type='password']:focus, &[type='email']:focus {
      box-shadow: ${(props: Object) => props.theme.auth.forms.inputs.boxShadow};
    }
  }

  & .remember label {
    font-weight: 500;
    margin-left: 5px;
    color: ${(props: Object) => props.theme.auth.forms.labels.color};
  }
`;

export const FormLogo = styled.div`
  text-align: center;
  margin-bottom: 20px;

  & img {
    max-width: 100%;
  }
`;

export const StyledForm = styled(Form)`
  padding: 30px 48px;
  background-color: ${(props: Object) => props.theme.auth.forms.bgColor};
`;

export const StyledFormTitle = styled.div`
  font-size: 18px;
  font-weight: bold;
  margin-bottom: 5px;;

  color: ${(props: Object) => props.theme.auth.forms.titleColor};
`;

export const StyledFormSubtitle = styled.div`
  font-size: 14px;
  margin-bottom: 21px;

  color: ${(props: Object) => props.theme.auth.forms.titleColor};
`;

export const StyledButton = styled(Button)`
  font-weight: bold;
  height: 42px;

  background-color: ${(props: Object) => props.theme.auth.forms.buttons.bgColor};
  color: ${(props: Object) => props.theme.auth.forms.buttons.textColor};
`;

export const StyledLinkTo = styled(LinkTo)`
  font-weight: 500;
  margin-right: 15px;

  color: ${(props: Object) => props.theme.auth.forms.labels.color};
`;

