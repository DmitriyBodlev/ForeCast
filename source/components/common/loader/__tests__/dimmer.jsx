import 'regenerator-runtime/runtime';

import React from 'react';
import { shallow } from 'enzyme';
import { DimmerComponent } from '../dimmer';
import { TableRectangle, TitleRectangle, Rectangle, DimmerWrapper } from '../ui';

describe('component/common/loader/dimmer.jsx', () => {
  it('should render component with wrapper and title sections', () => {
    const wrapper = shallow(<DimmerComponent />);

    const dimmerWrapperComponent = wrapper.find(DimmerWrapper);
    const titleRectangleComponent = wrapper.find(TitleRectangle);
    const tableRectangleComponent = wrapper.find(TableRectangle);
    const rectangleComponentInTable = tableRectangleComponent.find(Rectangle);
    const rectangleComponent = wrapper.find(Rectangle);

    expect(wrapper).toMatchSnapshot();
    expect(dimmerWrapperComponent.length).toBeGreaterThan(0);
    expect(titleRectangleComponent.length).toBeGreaterThan(0);
    expect(rectangleComponent.length).toBeGreaterThan(0);
    expect(tableRectangleComponent.length).toBe(0);
    expect(rectangleComponentInTable.length).toBe(0);
  });

  it('should render component with wrapper, title and table sections', () => {
    const wrapper = shallow(<DimmerComponent tableCount={2} rectangleCount={4} />);

    const dimmerWrapperComponent = wrapper.find(DimmerWrapper);
    const titleRectangleComponent = wrapper.find(TitleRectangle);
    const tableRectangleComponent = wrapper.find(TableRectangle);
    const rectangleComponentInTable = tableRectangleComponent.find(Rectangle);
    const rectangleComponent = wrapper.find(Rectangle);

    expect(wrapper).toMatchSnapshot();
    expect(dimmerWrapperComponent.length).toBeGreaterThan(0);
    expect(titleRectangleComponent.length).toBeGreaterThan(0);
    expect(rectangleComponent.length).toBeGreaterThan(0);
    expect(tableRectangleComponent.length).toBe(2);
    expect(rectangleComponentInTable.length).toBe(8);
  });
});
