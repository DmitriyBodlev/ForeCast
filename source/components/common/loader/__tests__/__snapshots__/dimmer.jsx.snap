// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`component/common/loader/dimmer.jsx should render component with user and admin sections 1`] = `
<WithTheme(styled(Component))>
  <WithTheme(styled(Component)) />
  <WithTheme(styled(Component)) />
  <WithTheme(styled(Component))>
    <WithTheme(styled(Component)) />
    <WithTheme(styled(Component)) />
    <WithTheme(styled(Component)) />
    <WithTheme(styled(Component)) />
  </WithTheme(styled(Component))>
  <WithTheme(styled(Component))>
    <WithTheme(styled(Component)) />
    <WithTheme(styled(Component)) />
    <WithTheme(styled(Component)) />
    <WithTheme(styled(Component)) />
  </WithTheme(styled(Component))>
</WithTheme(styled(Component))>
`;

exports[`component/common/loader/dimmer.jsx should render component with wrapper and content sections 1`] = `
<WithTheme(styled(Component))>
  <WithTheme(styled(Component)) />
  <WithTheme(styled(Component)) />
</WithTheme(styled(Component))>
`;

exports[`component/common/loader/dimmer.jsx should render component with wrapper and title sections 1`] = `
<WithTheme(styled(Component))>
  <WithTheme(styled(Component)) />
  <WithTheme(styled(Component)) />
</WithTheme(styled(Component))>
`;

exports[`component/common/loader/dimmer.jsx should render component with wrapper, title and table sections 1`] = `
<WithTheme(styled(Component))>
  <WithTheme(styled(Component)) />
  <WithTheme(styled(Component)) />
  <WithTheme(styled(Component))>
    <WithTheme(styled(Component)) />
    <WithTheme(styled(Component)) />
    <WithTheme(styled(Component)) />
    <WithTheme(styled(Component)) />
  </WithTheme(styled(Component))>
  <WithTheme(styled(Component))>
    <WithTheme(styled(Component)) />
    <WithTheme(styled(Component)) />
    <WithTheme(styled(Component)) />
    <WithTheme(styled(Component)) />
  </WithTheme(styled(Component))>
</WithTheme(styled(Component))>
`;

exports[`features/user-settings/index.jsx should render componecnt with user and admin sections 1`] = `
<WithTheme(styled(Component))>
  <WithTheme(styled(Component)) />
  <WithTheme(styled(Component)) />
</WithTheme(styled(Component))>
`;

exports[`features/user-settings/index.jsx should render component with user and admin sections 1`] = `
<WithTheme(styled(Component))>
  <WithTheme(styled(Component)) />
  <WithTheme(styled(Component)) />
  <WithTheme(styled(Component))>
    <WithTheme(styled(Component)) />
    <WithTheme(styled(Component)) />
    <WithTheme(styled(Component)) />
    <WithTheme(styled(Component)) />
  </WithTheme(styled(Component))>
  <WithTheme(styled(Component))>
    <WithTheme(styled(Component)) />
    <WithTheme(styled(Component)) />
    <WithTheme(styled(Component)) />
    <WithTheme(styled(Component)) />
  </WithTheme(styled(Component))>
</WithTheme(styled(Component))>
`;
